activitysynth/scripts/models.py

```diff
@@ -9,11 +9,123 @@
 from urbansim_templates import modelmanager as mm
 from urbansim_templates.models import LargeMultinomialLogitStep
 
+from activitysynth.scripts import utils
+
 
 # load existing model steps from the model manager
 mm.initialize()
 
 
+@orca.step()
+def impute_missing_skims(skims, beam_skims):
+    df = beam_skims.to_frame()
+    mtc = skims.to_frame(columns=['orig', 'dest', 'da_distance_AM'])
+    mtc.rename(
+        columns={'orig': 'from_zone_id', 'dest': 'to_zone_id'},
+        inplace=True)
+    mtc.set_index(['from_zone_id', 'to_zone_id'], inplace=True)
+
+    # miles to meters
+    mtc['dist'] = mtc['da_distance_AM'] * 1609.34
+
+    # create morning peak lookup
+    df['gen_time_per_m'] = df['gen_tt'] / df['distanceInM']
+    df['gen_cost_per_m'] = df['gen_cost'] / df['distanceInM']
+    df.loc[df['hour'].isin([7, 8, 9]), 'period'] = 'AM'
```

Comment:
I was using 6, 7, 8 as the AM peak because those hours are the most common in the skims. I think they mark the start of the hour, so 6, 7, 8 would represent 6am-8:59am.

Reply:
I meant to tell you, I found the following message in an old e-mail from Colin:

> We defined AM as 7-9 and PM as 3-5 in our Excerpt.